### DISCUSSION TITLE
Recommended hosting setups?

### DISCUSSION THREAD

**Discussion by jaredandrews** [wildstang/WildRank#204] [Q&A]
Hello,

My team has been testing WildRank and while we like it we have been having issues reliably hosting it. I was wondering if anyone would be willing to share how they host this app? Thanks!

**Comment by fruzyna** [comment]
The docker method is what we use to host the app: https://github.com/wildstang/WildRank/wiki/Execution However, there shouldn't be a significant difference between that and running outside a container. I'd imagine (but if I'm wrong please tell me otherwise) that the reliability issues you are having are the server going down or hanging when multiple users try to interact at the same time. The python webserver we use is very basic and not designed to handle concurrent users. The easiest way to avoid this problem is installing the PWA so that a local copy is cached on users devices therefore reaching out the server less frequently. This however requires hosting the server over https. The alternate option is running a local server on each individual device, which is the originally indented model when the server was built.

**Comment by jaredandrews** [comment]
Hey @fruzyna thanks for responding so quickly! I think the issue we are having is exactly what you describe. I have https set up now, so I'm thinking it will work better with everyone using PWA.

I'm not super familiar with running docker containers. Do you have a specific host you use / recommend? I would like to try that approach.

**Comment by fruzyna** [comment]
I just run our team's instance on my "server" (read: a retired desktop). The bottleneck is definitely in software right now so a Raspberry Pi could probably run it just fine.

As for using Docker it really is just 2 commands to get an instance up and running, once Docker is installed of course. I'm not as familiar with the Docker Desktop applications for Windows and Mac, but the provided docker-compose file should also be compatible with those applications.
